props/saber_caiwyn_buttons.h

```diff
@@ -0,0 +1,604 @@
+            if (SFX_mshuffle) {
+#ifndef DISABLE_BEEPS
+              beeper.Beep(0.1, 2000);
+#endif
+              hybrid_font.PlayPolyphonic(&SFX_mshuffle);
+            } else {
+              beeper.Beep(0.20,1000);
+              beeper.Beep(0.20,2000);
+              beeper.Beep(0.20,1000);
+            }
+            break;
+#endif
+          default:
+            track_mode_ = PLAYBACK_ONCE;
+            if (SFX_monce) {
+#ifndef DISABLE_BEEPS
+              beeper.Beep(0.1, 2000);
+#endif
+              hybrid_font.PlayPolyphonic(&SFX_monce);
+            } else {
+              beeper.Beep(0.20,1000);
+              beeper.Beep(0.20,2000);
+              beeper.Beep(0.20,1000);
+            }
+        }
+        return true;
+
+// Non-Impact Clash and Lockup / Zoom Color
+      case EVENTID(BUTTON_POWER, EVENT_PRESSED, MODE_ON):
+        if (!SaberBase::Lockup() && !mode_volume_ && (SaberBase::GetColorChangeMode() == SaberBase::COLOR_CHANGE_MODE_NONE)) {
+          SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+          SaberBase::DoBeginLockup();
+#ifndef DISABLE_BUTTON_CLASH
+          hybrid_font.PlayPolyphonic(&SFX_clsh);
+#endif
+          return true;
+#ifndef DISABLE_COLOR_CHANGE
+        } else if (SaberBase::GetColorChangeMode() == SaberBase::COLOR_CHANGE_MODE_SMOOTH) {
+#ifndef DISABLE_BEEPS
+          if (SFX_ccend) {
+            beeper.Beep(0.1, 2000);
+          }
+#endif
+          SaberBase::SetColorChangeMode(SaberBase::COLOR_CHANGE_MODE_ZOOMED);
+          return true;
+#endif;
+        }
+        break;
+
+// Save Volume / Save Color
+      case EVENTID(BUTTON_POWER, EVENT_CLICK_SHORT, MODE_ON):
+        if (mode_volume_) {
+          mode_volume_ = false;
+          if (SFX_vmend) {
+#ifndef DISABLE_BEEPS
+            beeper.Beep(0.1, 2000);
+#endif
+            SFX_vmend.Select(0);
+            hybrid_font.PlayPolyphonic(&SFX_vmend);
+          } else {
+            beeper.Beep(0.20,2000.0);
+            beeper.Beep(0.20,1414.2);
+            beeper.Beep(0.20,1000.0);
+          }
+          STDOUT.println("Exit Volume Menu");
+          return true;
+#ifndef DISABLE_COLOR_CHANGE
+        } else if (SaberBase::GetColorChangeMode() == SaberBase::COLOR_CHANGE_MODE_STEPPED) {
+          if (SFX_ccend) {
+#ifndef DISABLE_BEEPS
+            beeper.Beep(0.1, 2000);
+#endif
+            SFX_ccend.Select(0);
+          }
+          ToggleColorChangeMode();
+          return true;
+#endif
+        }
+        break;
+
+#ifndef DISABLE_COLOR_CHANGE
+      case EVENTID(BUTTON_POWER, EVENT_RELEASED, MODE_ON):
+        if (SaberBase::GetColorChangeMode() == SaberBase::COLOR_CHANGE_MODE_ZOOMED) {
+          if (SFX_ccend) {
+            SFX_ccend.Select(0);
+          }
+          ToggleColorChangeMode();
+          return true;
+        }
+        break;
+#endif
+
+// Cancel Volume / Reset Color / Blaster Block
+      case EVENTID(BUTTON_AUX, EVENT_CLICK_SHORT, MODE_ON):
+        if (mode_volume_) {
```

Comment:
In general, all of these large blocks of code should not go in this function, but in helper functions.

Reply:
That was just a stylistic decision.  I didn't put anything into a helper function that didn't need to be repeated.

Reply:
Done.